Drivers/c/PioHwas/test/main.cpp

```diff
@@ -0,0 +1,398 @@
+                                    const PioHwas_Pin_Config &configPioHwas) {
+
+  switch (configPioHwas.filter) {
+  case PioHwas_Filter_Debounce:
+    configPio.filter = Pio_Filter_Debounce;
+    break;
+  case PioHwas_Filter_Glitch:
+    configPio.filter = Pio_Filter_Glitch;
+    break;
+  case PioHwas_Filter_None:
+    configPio.filter = Pio_Filter_None;
+    break;
+  }
+}
+
+static inline void ConvertPioPull(Pio_Pin_Config &configPio,
+                                  const PioHwas_Pin_Config &configPioHwas) {
+
+  switch (configPioHwas.pull) {
+  case PioHwas_Pull_Down:
+    configPio.pull = Pio_Pull_Down;
+    break;
+  case PioHwas_Pull_Up:
+    configPio.pull = Pio_Pull_Up;
+    break;
+  case PioHwas_Pull_None:
+    configPio.pull = Pio_Pull_None;
+    break;
+  }
+}
+
+static inline Pio_Port GetPioPort(PioHwas_Port port) {
+  switch (port) {
+  case PioHwas_Port_A:
+    return Pio_Port_A;
+  case PioHwas_Port_B:
+    return Pio_Port_B;
+  case PioHwas_Port_C:
+    return Pio_Port_C;
+  case PioHwas_Port_D:
+    return Pio_Port_D;
+  case PioHwas_Port_E:
+    break;
+  }
+  return Pio_Port_E;
+}
+
+static inline Pio_Pin_Config
+ConvertConfigPioHwasToConfigPio(PioHwas_Pin_Config &configPioHwas) {
+  Pio_Pin_Config configPio;
+
+  ConvertPioDirection(configPio, configPioHwas);
+  ConvertPioFilter(configPio, configPioHwas);
+  ConvertPioPull(configPio, configPioHwas);
+
+  configPio.control = Pio_Control_Pio;
+  configPio.irq = Pio_Irq_None;
+  configPio.isMultiDriveEnabled = false;
+  configPio.isSchmittTriggerDisabled = true;
+  return configPio;
+}
+
+bool operator==(Pio_Pin_Config &lhs, Pio_Pin_Config &rhs) {
+  LONGS_EQUAL(lhs.control, rhs.control);
+  LONGS_EQUAL(lhs.direction, rhs.direction);
+  LONGS_EQUAL(lhs.filter, rhs.filter);
+  LONGS_EQUAL(lhs.irq, rhs.irq);
+  LONGS_EQUAL(lhs.isMultiDriveEnabled, rhs.isMultiDriveEnabled);
+  LONGS_EQUAL(lhs.isSchmittTriggerDisabled, rhs.isSchmittTriggerDisabled);
+  LONGS_EQUAL(lhs.pull, rhs.pull);
+  return true;
+}
+
+static void VerifyConfig(const PioHwas &pioHwas,
+                         PioHwas_Pin_Config &configPioHwas) {
+  Pio_Pin_Config configPioExpected =
+      ConvertConfigPioHwasToConfigPio(configPioHwas);
+  Pio_Pin_Config configPioActual;
+  Pio pio;
+  Pio_init(GetPioPort(configPioHwas.port), &pio);
+
+  int errorCode{0};
+  Pio_getPinsConfig(&pio, static_cast<const uint32_t>(pioHwas.pin),
+                    &configPioActual, &errorCode);
+
+  LONGS_EQUAL(0, errorCode);
+  CHECK_TRUE(configPioExpected == configPioActual);
+}
+
+static inline void Verify_PioHwas_init_pin(const PioHwas &testPio,
+                                           PioHwas_Pin_Config &config) {
+  VerifyPmc(config.port);
+  VerifyPioPsr(testPio);
+  VerifyConfig(testPio, config);
+}
+
+TEST_GROUP(PioHwas_init_pin) {
+  PioHwas testPio;
+  PioHwas_Pin_Config config;
+
+  void teardown() override {
+    Verify_PioHwas_init_pin(testPio, config);
```

Comment:
Why is there a verify in teardown?

Reply:
Because all tests in the group have the same verification procedure. Instead of calling this function at the end of each test, I call it in the "teardown" function.